pkg/controller/manager/nad/controller.go

```diff
@@ -125,6 +135,43 @@ func (h Handler) OnRemove(key string, nad *cniv1.NetworkAttachmentDefinition) (*
 	return nad, nil
 }
 
+func (h Handler) ensureLabels(nad *cniv1.NetworkAttachmentDefinition) error {
+	if nad.Labels != nil && nad.Labels[utils.KeyNetworkType] != "" && nad.Labels[utils.KeyClusterNetworkLabel] != "" {
+		return nil
+	}
+
+	labels := nad.Labels
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	netconf := &utils.NetConf{}
+	if err := json.Unmarshal([]byte(nad.Spec.Config), netconf); err != nil {
+		return fmt.Errorf("unmarshal config of nad %s/%s failed, error: %w", nad.Namespace, nad.Name, err)
```

Comment:
the error message looks overlapped with #106, we can return the error directly here.